core/src/main/java/org/n52/iceland/ogc/filter/ComparisonFilter.java

```diff
@@ -0,0 +1,315 @@
+
+    /**
+     * wild card character
+     */
+    private String wildCard;
+
+    /**
+     * single char character
+     */
+    private String singleChar;
+
+    private boolean matchCase = true;
+
+    /**
+     * default constructor
+     */
+    public ComparisonFilter() {
+    }
+
+    /**
+     * constructor
+     *
+     * @param operator
+     *            Filter operator
+     * @param valueReference
+     *            valueReference
+     * @param value
+     *            value
+     */
+    public ComparisonFilter(ComparisonOperator operator, String valueReference, String value) {
+        super(valueReference);
+        this.operator = operator;
+        this.value = value;
+    }
+
+    /**
+     * constructor for {@link ComparisonOperator#PropertyIsBetween} filter
+     *
+     * @param operator
+     *            Filter operator
+     * @param valueReference
+     *            valueReference
+     * @param value
+     *            value
+     * @param valueUpper
+     *            upper value
+     * @throws OwsExceptionReport
+     *             If operator is not
+     *             {@link ComparisonOperator#PropertyIsBetween}
+     */
+    public ComparisonFilter(ComparisonOperator operator, String valueReference, String value, String valueUpper)
+            throws OwsExceptionReport {
+        super(valueReference);
+        if (operator == ComparisonOperator.PropertyIsBetween) {
+            this.operator = operator;
+            this.value = value;
+            this.valueUpper = valueUpper;
+        } else {
+            throw new NoApplicableCodeException()
+                    .withMessage("Use other constructor for ComparisonFilter! This constructor could only"
+                            + "be used for operator 'PropertyIsBetween'");
+        }
+    }
+
+    /**
+     * constructor for {@link ComparisonOperator#PropertyIsLike} filter
+     *
+     * @param operator
+     *            Filter operator
+     * @param valueReference
+     *            valueReference
+     * @param value
+     *            value
+     * @param valueUpper
+     *            upper value for between filter
+     * @param escapeString
+     *            Escape characters
+     * @throws OwsExceptionReport
+     *             If operator is not {@link ComparisonOperator#PropertyIsLike}
+     */
+    public ComparisonFilter(ComparisonOperator operator, String valueReference, String value, String valueUpper,
+            String escapeString) throws OwsExceptionReport {
+        super(valueReference);
+        if (operator == ComparisonOperator.PropertyIsLike) {
+            this.operator = operator;
+            this.value = value;
+            this.valueUpper = valueUpper;
+            this.escapeString = escapeString;
+        } else {
+            throw new NoApplicableCodeException()
+                    .withMessage("Use other constructor for ComparisonFilter! This constructor could only be used for operator 'PropertyIsLike'");
+        }
```

Comment:
Since this will have a considerable input on the existing parsers, @CarstenHollmann and I will have to coordinate on the refactoring > https://github.com/52North/iceland/issues/37